org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/HandleCryptosystemTextbook.java

```diff
@@ -0,0 +1,974 @@
+				cstb.txtThirdPlaintext.setText("");
+				cstb.txtFourthPlaintext.setText("");
+			}
+			
+			
+			
+			if(!cstb.rftc.txtModN.getText().isEmpty() && !cstb.textSelector.getText().getText().isEmpty()) {
+				if(newTextselectorText != oldTextselectorTextDecryptionMode) {
+					cstb.btnDecryption.setEnabled(true);
+				}
+				else {
+					if(cstb.txtCiphertextSegments.getText().isEmpty())
+						cstb.btnDecryption.setEnabled(true);
+					else
+						cstb.btnDecryption.setEnabled(false);
+				}
+			}
+			else {
+				cstb.btnDecryption.setEnabled(false);
+			}
+		}
+		
+		
+		/*if(cstb.btnRadioDecrypt.getSelection()) {
+			cstb.txtCiphertextSegments.setText("");
+			if(!cstb.rftc.txtModN.getText().isEmpty() && cstb.textSelector.getText() != null && !cstb.textSelector.getText().getText().isEmpty()) {
+				String pattern = "[0-9a-fA-F]+";
+				
+				if(!cstb.textSelector.getText().getText().matches(pattern)) {
+					cstb.btnDecryption.setEnabled(false);
+					return;
+				}
+				
+				String newTextselectorText = cstb.textSelector.getText().getText();
+				if(newTextselectorText != oldTextselectorTextDecryptionMode) {
+					cstb.btnDecryption.setEnabled(true);
+				}
+				else {
+					if(cstb.txtCiphertextSegments.getText().isEmpty())
+						cstb.btnDecryption.setEnabled(true);
+					else
+						cstb.btnDecryption.setEnabled(false);
+				}
+			}
+			else {
+				cstb.btnDecryption.setEnabled(false);
+			}
+				
+		}*/
+	}
+	
+	
+	public void textSelectorAction(CryptosystemTextbookComposite cstb) {
+		if(cstb.btnRadioEncrypt.getSelection()) {
+			
+			if(cstb.textSelector.getText() != null) {
+				this.hideControl(cstb.txtLoadTextWarning);
+				
+				String newTextselectorText = null;
+				
+				// maybe uncomment again, atm its just testing
+				/*if(firstTextselectorAccessEncryptionMode) {
+					oldTextselectorTextEncryptionMode = cstb.textSelector.getText().getText();
+					firstTextselectorAccessEncryptionMode = true;
+					return;
+				}*/
+				
+				newTextselectorText = cstb.textSelector.getText().getText();
+				if(newTextselectorText != oldTextselectorTextEncryptionMode) {
+					cstb.txtCiphertext.setText("");
+					cstb.txtPlaintext.setText("");
+				}
+				
+				
+				
+				if(!cstb.rftc.txtModN.getText().isEmpty() && !cstb.textSelector.getText().getText().isEmpty()) {
+					cstb.cmbChooseBlockPadding.setEnabled(true);
+					if(newTextselectorText != oldTextselectorTextEncryptionMode) {
+						cstb.btnEncrypt.setEnabled(true);
+					}
+					else {
+						if(cstb.txtCiphertext.getText().isEmpty())
+							cstb.btnEncrypt.setEnabled(true);
+						else {
+							cstb.btnEncrypt.setEnabled(false);
+						}
+					}
+				}
+				else {
+					cstb.cmbChooseBlockPadding.setEnabled(false);
+					cstb.btnEncrypt.setEnabled(false);
+					
+					if(cstb.textSelector.getText().getText().isEmpty()) {
+						cstb.txtLoadTextWarning.setText("Attention: load a non-empty plaintext");
+						this.showControl(cstb.txtLoadTextWarning);
+					}
+				}
```

Comment:
Those are way too many if's stacked. No one is ever gonna figure out what this code is doing.
Rather split it up into nice little methods: